nova/compute/api.py

```diff
@@ -742,6 +769,9 @@ def not_image_and_root_bdm(bdm):
         block_device_mapping = self._merge_bdms_lists(
             image_defined_bdms, block_device_mapping)
 
+        if self._flavor_enforces_boot_from_volume(instance_type):
+            self._enforce_boot_from_volume(image_meta, instance_type,
+                                           block_device_mapping)
```

Comment:
Doesn't the above-called `block_device.from_legacy_mapping()` already add the image-bdm? https://github.com/sapcc/nova/blob/5b052045c355b1ecba3b3b0e1116b9092f1fd1ce/nova/block_device.py#L355-L356
The flavor would then be unusable ...

Reply:
Oh, right. This worked well initially, when the image mapping was silently overwritten to a volume mapping.

But now since we throw an error, it will indeed make it unusable. I'll revisit this.

Reply:
Please have another look. Thanks.